hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -506,12 +487,11 @@ static void add_final_stage_code(struct PixelShader *ps, struct FCInputInfo fina
     QString *d = get_input_var(ps, final.d, false);
     QString *g = get_input_var(ps, final.g, false);
 
-    add_var_ref(ps, "r0");
-    qstring_append_fmt(ps->code, "r0.rgb = %s + mix(vec3(%s), vec3(%s), vec3(%s));\n",
+    qstring_append_fmt(ps->code, "fragColor.rgb = %s + mix(vec3(%s), vec3(%s), vec3(%s));\n",
```

Comment:
Why change this here? Don't we set `fragColor = r0` later on?

Reply:
Using `r0` etc. is a leftover from Cxbx code where they used D3D ASM shaders. So they had to use these sort of temporaries.

The real output variable is called `fcoc` ([See "3.8.12.3  Final Combiner Stage Operation"](https://www.khronos.org/registry/OpenGL/extensions/NV/NV_register_combiners.txt)).

`fragColor` is much closer and shows the intention.

*I might rename `fragColor` to `fcoc` in the future or add a temporary again (for other reasons), but it's not really beneficial here. Naming it `r0` is just confusing.*